### DISCUSSION TITLE
How to reproduce the vanilla generation

### DISCUSSION THREAD

**Discussion by ghost** [Minestom/Minestom#1156] [Help]
I search in Minestom a way to reproduce the vanilla generation but i have no idea for the repartition of biome and how to reproduce the relief of a mountain or a just plains. If someone have an idea i take notes ^^

**Comment by MrGazdag** [comment]
This has been asked before, but the main answer is either:
 - reimplement vanilla world generation yourself (it will take a long time)
 - use a vanilla/spigot/paper/whatever server that generates worlds, then copy the chunks over to Minestom (this has been done already, check the #showcase channel on Discord)

**Comment by Flopgop** [comment]
If you choose to reimplement it yourself a great place to start would be watching [this video from Henrik Kniberg explaining the world generation](https://www.youtube.com/watch?v=CSa5O6knuwI) and if you need any specific values (eg. spline values, offsets etc) you can go to [this repo](https://github.com/slicedlime/examples) on slicedlimes github for the vanilla generation values. I recommend using [FastNoiseLite](https://github.com/Auburn/FastNoiseLite) for your noise library and [Apache Commons Math](https://mvnrepository.com/artifact/org.apache.commons/commons-math3) for spline interpolators/functions
